pynamodb/pagination.py

```diff
@@ -100,7 +100,39 @@ def next(self):
 
     @property
     def last_evaluated_key(self):
-        return self.page_iter.last_evaluated_key
+        if self._first_iteration:
+            # Not started iterating yet: there cannot be a last_evaluated_key
+            return None
+
+        if self._index == self._count:
+            # Entire page has been consumed: last_evaluated_key is whatever DynamoDB returned
+            # It may correspond to the current item, or it may correspond to an item evaluated but not returned.
+            return self.page_iter.last_evaluated_key
+
+        # In the middle of a page of results: reconstruct a last_evaluated_key from the current item
+        # The operation should be resumed starting at the last item returned, not the last item evaluated.
+        # This can occur if the 'limit' is reached in the middle of a page.
+        item = self._items[self._index - 1]
+        last_evaluated_key = {}
+        if self.page_iter.last_evaluated_key:
+            # If the current page has a last_evaluated_key, use it to determine key attributes
+            for key in self.page_iter.last_evaluated_key.keys():
+                last_evaluated_key[key] = item[key]
+        else:
+            # Use the table meta data to determine the key attributes
+            table_connection = self.page_iter._operation.im_self
```

Comment:
should we write a private method to expose this? do we need this other places in the file?